[Keyboard] Add QAZ Shield

tominabox1 opened the pull request:
Add QAZ shield

petejohanson commented on app/boards/shields/qaz/qaz.conf:
```diff
@@ -0,0 +1,26 @@
+# Use the STRIP config specific to the LEDs you're using
+CONFIG_WS2812_STRIP=n
+# Turn on logging, and set ZMK logging to debug output
+CONFIG_LOG=y
+CONFIG_ZMK_LOG_LEVEL_DBG=y
+
+# Turn on USB CDC ACM device
+CONFIG_USB=y
+CONFIG_USB_DEVICE_STACK=y
+CONFIG_USB_CDC_ACM=y
+CONFIG_USB_CDC_ACM_RINGBUF_SIZE=1024
+CONFIG_USB_CDC_ACM_DEVICE_NAME="CDC_ACM"
+CONFIG_USB_CDC_ACM_DEVICE_COUNT=1
+
+# Enable serial console
+CONFIG_SERIAL=y
+CONFIG_CONSOLE=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_UART_LINE_CTRL=y
+
+# Enable USB UART, and set the console device
+CONFIG_UART_CONSOLE=y
+CONFIG_USB_UART_CONSOLE=y
+CONFIG_UART_CONSOLE_ON_DEV_NAME="CDC_ACM_0"
+CONFIG_USB_UART_DTR_WAIT=n
```
We have *not* been turning on logging by default in these files, since it is a theoretically security concern, since it logs keystroke information. These should be removed for the version of this for in-tree use.

tominabox1 replied:
oops forgot I had that in there

petejohanson commented on app/boards/shields/qaz/qaz.keymap:
```diff
@@ -0,0 +1,75 @@
+			bindings = <&bootloader>, <&reset>, <&kp BKSP>;
+        };
+
+        num_spc: behavior_tap_hold2 {
```
Tap hold is not in `main`, so this should not be in here. Once we merge that, this can be restored, but for now this should be in your personal user config keymap for this.

petejohanson commented on app/boards/shields/qaz/qaz.keymap:
```diff
@@ -0,0 +1,75 @@
+#include <behaviors.dtsi>
```
Please add a correct copyright + license header here. Thanks!

petejohanson replied:
Just this one file still missing a license header.

petejohanson commented on app/boards/shields/qaz/qaz.overlay:
```diff
@@ -0,0 +1,48 @@
+    chosen {
+        zmk,kscan = &kscan0;
+        zmk,matrix_transform = &default_transform;
+        zmk,underglow = &led_strip;
```
This should be added in the nice_nano overlay where the led_strip is defined. This will ensure this shield builds properly with other controllers like the BlueMicro840.

petejohanson commented on app/boards/shields/qaz/qaz.overlay:
```diff
@@ -0,0 +1,48 @@
+#include <dt-bindings/zmk/matrix-transform.h>
```
Please add a license & copyright header here as well. Thanks!

petejohanson commented on app/boards/shields/qaz/Kconfig.shield:
```diff
@@ -0,0 +1,2 @@
+config SHIELD_MY_BOARD
```
Please add a copyright/license header. Thanks!

petejohanson commented on app/boards/shields/qaz/Kconfig.defconfig:
```diff
@@ -0,0 +1,6 @@
+if SHIELD_QAZ
```
Please add a copyright/license header. Thanks!

tominabox1 commented:
Updated as requested. I removed RGB support. I have not been able to get it to work on my board as I believe we are using one of the low freq pins (P1.15) so may not be possible.
